content/guides/docker-concepts/running-containers/persisting-container-data.md

```diff
@@ -0,0 +1,176 @@
+
+As an example, imagine you create a volume named `log-data`. When starting a container with the following command, the volume will be mounted (or attached) into the container at `/logs`:
+
+```console
+$ docker run -v log-data:/logs my-image
+```
+
+When the container runs, all files it writes into the `/logs` folder will be saved in this volume, outside of the container. If you delete the container and start a new container using the same volume, the files will still be there.
+
+> **Sharing files using volumes**
+>
+> As a pro-tip, you can attach the same volume to multiple containers to easily share files between containers. This might be helpful in scenarios such as log aggregation, data pipelines, or other event-driven applications.
+{ .tip }
+
+
+### Managing volumes
+
+Volumes have their own lifecycle beyond that of containers and can grow quite large depending on the type of data and applications you’re using. The following commands will be helpful to manage volumes:
+
+- `docker volume ls` - list all volumes
+- `docker volume rm <volume-name-or-id>` - remove a volume (only works when the volume is not attached to any containers)
+- `docker volume prune` - remove all unused (unattached) volumes
+
+> **Auto-creation of volumes**
+> 
+> While you can use the `docker volume create` command to create a volume, Docker will automatically create a volume when needed. For example, the following command will auto-create a volume named `sample-volume` if it doesn’t exist:
+>
+> ```console
+> $ docker run -v sample-volumes:/data my-image
+> ```
+
+
+## Try it out
+
+In this guide, you’ll practice creating and using volumes to persist data created by a Postgres container. When the database runs, it stores files into the `/var/lib/postgresql/data` directory. By attaching the volume here, you will be able to restart the container multiple times while keeping the data.
+
+### Use volumes
+
+1. [Download and instal](https://www.docker.com/products/docker-desktop/)l Docker Desktop.
+
+2. Start a container using the [Postgres image](https://hub.docker.com/_/postgres) with the following command:
+
+    ```console
+    $ docker run --name=db -e POSTGRES_PASSWORD=secret -d -v postgres_data:/var/lib/postgresql/data postgres
+    ```
+
+    This will start the database in the background, configure it with a password, and attach a volume to the directory PostgreSQL will persist the database files.
+
+3. Connect to the database by using the following command:
+
+    ```console
+    $ docker exec -ti db psql -U postgres
+    ```
+
+4. In the PostgreSQL command line, run the following to create a database table and insert two records:
+
+    ```psql
+    CREATE TABLE tasks (
+        id SERIAL PRIMARY KEY,
+        description VARCHAR(100)
+    );
+    INSERT INTO tasks (description) VALUES ('Finish work'), ('Have fun');
+    ```
+
+5. Verify the data is in the database by running the following in the PostgreSQL command line:
+
+    ```psql
```

Comment:
The formatting doesn't look great. Maybe consider using `text` instead of `psql`.

Reply:
Fixed it. Can you check it once?